src/categorical_algebra/DPO.jl

```diff
@@ -0,0 +1,215 @@
+module DPO
+export rewrite, rewrite_match, valid_dpo, dangling_condition, id_condition, pushout_complement, extend_cset,subcset
+
+using ..FinSets
+import ..CSets: AbstractACSet, ACSet, ACSetTransformation, components, homomorphism, homomorphisms, unpack_components, pack_components, add_parts!, nparts, set_subpart!, is_natural
+using ...Theories
+using ..Limits
+
+"""
+    l
+  L ← I
+m ↓   ↓k
+  G ← K
+    g
+
+Compute a pushout complement, componentwise in Set. On a formal level:
+For each component, define K = G / m(L/l(I)). There is a natural injection g: K↪G
+For each component, define k as equal to the map l;m (every element in the image in G is also in K).
+
+Returns ACSetTransformations k and g such that (m, g) is the pushout of (l, k)
+
+Implementation-wise, elements of K are ordered in the same order as they appear in G.
+Construct an offset mapping to keep track of the location of elements of G within K.
+"""
+function pushout_complement(l::ACSetTransformation,m::ACSetTransformation)::Pair{ACSetTransformation,ACSetTransformation}
```

Comment:
`l` being monic guarantees uniqueness, but I haven't seen much in the literature for the alternative. E.g. [this](https://core.ac.uk/download/pdf/82404673.pdf) paper merely restricts to monic `l` due to "ease of construction" and because it's "sufficient".